Python/symtable.c

```diff
@@ -1078,7 +1077,7 @@ analyze_block(PySTEntryObject *ste, PyObject *bound, PyObject *free,
         goto error;
     /* Records the results of the analysis in the symbol table entry */
     if (!update_symbols(ste->ste_symbols, scopes, bound, newfree, inlined_cells,
-                        ste->ste_type == ClassBlock))
+                        (ste->ste_type == ClassBlock) || class_entry))
```

Comment:
```suggestion
                        (ste->ste_type == ClassBlock) || ste->ste_can_see_class_scope))
```
This is equivalent but doesn't involve casting a pointer to an int.

Reply:
Is it equivalent? What about annotation scopes that do not have an enclosing class scope; will they still have can_see_class_scopes set? I'm not sure we want to apply this if there isn't an enclosing class scope.

Reply:
Yes, `ste_can_see_class_scope` is set only for annotation blocks within classes. I made this change locally and `test_type_params` passed.